src/redux/sagas/utils/pollAction.ts

```diff
@@ -0,0 +1,42 @@
+import { ApolloQueryResult } from '@apollo/client';
+import { call, delay, put, race, take } from 'redux-saga/effects';
+
+import { ContextModule, getContext } from '~context';
+import { GetColonyActionDocument, GetColonyActionQuery } from '~gql';
+import { ActionTypes } from '~redux/actionTypes';
+
+const POLLING_DELAY_IN_MS = 3000;
+
+/**
+ * Util function polling Amplify to check whether an action with a given transaction hash
+ * has already been processed
+ */
+export function* waitForIngestorToHandleAction(transactionHash: string) {
+  yield race({
+    task: call(pollAction, transactionHash),
+    cancel: take(ActionTypes.CANCEL_ACTION_POLLING),
+  });
+}
+
+function* pollAction(transactionHash: string) {
+  while (true) {
+    try {
+      const apolloClient = getContext(ContextModule.ApolloClient);
+      const result = (yield apolloClient.query({
+        query: GetColonyActionDocument,
+        variables: {
+          transactionHash,
+        },
+        fetchPolicy: 'network-only',
+      })) as ApolloQueryResult<GetColonyActionQuery>;
+
+      if (result.data.getColonyAction) {
+        yield put({ type: ActionTypes.CANCEL_ACTION_POLLING });
```

Comment:
I'd argue we would want to differentiate actions names if the poll action succeeds vs. it fails.

Just from a debug/logical standpoint _(since the actions don't actually do anything, other than being dispatched)_

But it would still be nice to have something like `SUCCESS_ACTION_POLLING` on success and `CANCEL_ACTION_POLLING` on fail / cancel